File: Python PowerPoint Examples/ShapesApiDemo/setSlideShapeInfo.py
import spirecloudpowerpoint
from spirecloudpowerpoint.configuration import *

appId = "your id"
appKey = "your key"
baseUrl = "https://api.e-iceblue.cn"
configuration = Configuration(appId, appKey,baseUrl)
api= spirecloudpowerpoint.api.shapes_api.ShapesApi(configuration)

fileName = "SetSlideShapeInfo.pptx"
inputFolder = "input"
password = ""
storage = ""
slide_index = 0
shape_index = "3"
source = spirecloudpowerpoint.models.ShapeBase()
source.shape_type = "Rectangle"
source.x = 50
source.y = 50
source.height = 200
source.width = 100
source.alternative_text = "AlternativeText"
source.z_order_position = 0
source.type = "ShapeBase"
source.fill_format = spirecloudpowerpoint.models.SolidFill("red")
source.z_order_position = 1
result =api.set_slide_shape_info(fileName, slide_index=slide_index, shape_index=shape_index,source=source,password=password, storage=storage,folder=inputFolder)

File: JavaScript PowerPoint Examples/ShapesApiDemo/setSlideShapeInfo.js
(function (){
    var appId = "your ID";
    var appKey = "your Key";
    var baseUrl = "https://api.e-iceblue.cn";

    var SpirecloudPowerpoint = require('../../src/index');
    var apiClient = new SpirecloudPowerpoint.ApiClient(appId, appKey,baseUrl);
    var instance = new SpirecloudPowerpoint.ShapesApi(apiClient);
    
    var name = 'SetSlideShapeInfo.pptx';
    var folder = 'input';
    var slideIndex = 0;
    var shapeIndex = 2;
    var opts = {};
    opts.source = new SpirecloudPowerpoint.ShapeBase();
    opts.source.ShapeType = "Rectangle";
    opts.source.X = 50;
    opts.source.Y = 50;
    opts.source.Height = 200;
    opts.source.Width = 100;
    opts.source.AlternativeText = "AlternativeText";
    opts.source.FillFormat = new SpirecloudPowerpoint.SolidFill("Red");
    opts.source.FillFormat.Type = "Solid";
    opts.source.ZorderPosition = 1;
    opts.folder = folder;
    var response = instance.setSlideShapeInfo(name, slideIndex, shapeIndex, opts,
     function (error, data, response) {
        if (error) throw error;
     });
})();